### PR TITLE
Feat/remove view type use programme slug - LESQ-208

### PR DESCRIPTION
**Pull request by JBR90**
## Description

- Api fetches curriculumApi / curriculumApi2023 are now dependant on '-l' suffix on programmeSlug or subjectSlug on the programme listing page
- downloads refactored to use programmeSlug suffix for vercelApi / downloadsApi
- avo - get page view now from path not. viewType
- update tests

## Acceptance Criteria:

See notion doc: [[Combining 2020/2023 data in same experience](https://www.notion.so/Combining-2020-2023-data-in-same-experience-2935790c128b48c0906b2390637cc82d?pvs=21)](https://www.notion.so/Combining-2020-2023-data-in-same-experience-2935790c128b48c0906b2390637cc82d?pvs=21) ← this is the nextjs approach

- [x]  Write helper functions to append and remove ‘-l’
- [x]  Transform slug in sensible and consistent way (api and search)
- [x]  In any page within `/beta/teachers/programmes/{programmeSlug}/*` , in `getStaticProps` , check to see if it’s a '-l' slug, if so, fetch from the 2020 api, otherwise fetch from the 2023 api
- [x]  Remove view types, everything should exist at /teachers

## Issue(s)

Fixes #

## How to test
Test the lesson browsing journey is working as expected from /teachers

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner

### CODE REVIEW

File: src/__tests__/pages/beta/[viewType]/programmes/[programmeSlug]/units/unitListing.test.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import curriculumApi from "../../../../../../../node-lib/curriculum-api/__mocks_
 import UnitListingPage, {
   getStaticPaths,
   getStaticProps,
-} from "../../../../../../../pages/[viewType]/programmes/[programmeSlug]/units";
+} from "../../../../../../../pages/teachers/programmes/[programmeSlug]/units";
```
Comment (kimon-satan) [inline src/__tests__/pages/beta/[viewType]/programmes/[programmeSlug]/units/unitListing.test.tsx:7]: please start replacing these with aliases

File: src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/downloads.test.tsx [hunk 2]
```diff
@@ -5,23 +5,24 @@ import userEvent from "@testing-library/user-event";
 import { computeAccessibleDescription } from "dom-accessibility-api";
 import React from "react";
 
+import removeLegacySlugSuffix from "../../../../../../../../../utils/slugModifiers/removeLegacySlugSuffix";
```
Comment (kimon-satan) [inline src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/downloads.test.tsx:8]: don't introduce new relative paths. Use aliases

**Review by kimon-satan — APPROVED** [review]

File: src/__tests__/pages/beta/[viewType]/programmes/[programmeSlug]/units/unitListing.test.tsx (hunk 1 above)
Comment (JBR90) [reply src/__tests__/pages/beta/[viewType]/programmes/[programmeSlug]/units/unitListing.test.tsx:7]: Have done for a lot of pages in this PR, missed this one

File: src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/downloads.test.tsx (hunk 2 above)
Comment (JBR90) [reply src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/downloads.test.tsx:8]: Will do, sometimes hard to spot when it auto imports

File: src/utils/slugModifiers/removeLegacySlugSuffix.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import isSlugLegacy from "./isSlugLegacy";
+
+const removeLegacySlugSuffix = (slug: string) => {
+  return isSlugLegacy(slug) ? slug.slice(0, -2) : slug;
```
Comment (mantagen) [inline src/utils/slugModifiers/removeLegacySlugSuffix.ts:4]: Again, best not to have -2 being fixed i'd say.
Either `slug.slice(0, -(LEGACY_PROGRAMME_SUFFIX.length))` (if that works)
or `const [firstPart] = slug.split(LEGACY_PROGRAMME_SUFFIX)`

File: src/utils/slugModifiers/isSlugLegacy.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export const LEGACY_PROGRAMME_SUFFIX = "-l";
```
Comment (mantagen) [inline src/utils/slugModifiers/isSlugLegacy.ts:1]: Perhaps this could just be `LEGACY_SLUG_SUFFIX` because it's used for both programme and subject

**Review by jack-skerrett-oak — APPROVED** [review]
Works nicely. Haven't seen any issues when navigate around the site.

**Pull request merged**